components/Controllers/Activity/Activity.tsx

```diff
@@ -9,138 +9,106 @@ import { Fieldset } from 'components/Fieldset';
   account?: string;
   // If scoping activity view to just a specific vault
   vault?: string;
   showSwaps?: boolean;
-  subgraphPool: NonNullable<PoolByIdQuery['pool']>;
 };
 
-const EVENT_INCREMENT = 5;
+const LIMIT = 5;
+
+function activityIsSwap(activity: ActivityType) {
+  return !!activity.amountIn && !!activity.amountOut;
+}
+
+function activityIsAddCollateral(activity: ActivityType) {
+  return activity.addedCollateral.length > 0 || !!activity.amountBorrowed;
+}
+
+function activityIsRemoveCollateral(activity: ActivityType) {
+  return activity.removedCollateral.length > 0 || !!activity.amountRepaid;
+}
+
+function activityIsAuctionStart(activity: ActivityType) {
+  return !!activity.auctionTokenId && !activity.auctionEndPrice;
+}
 
-export function Activity({
-  account,
-  vault,
-  showSwaps = true,
-  subgraphPool,
-}: ActivityProps) {
+function activityIsAuctionEnd(activity: ActivityType) {
+  return !!activity.auctionTokenId && !!activity.auctionEndPrice;
+}
+
+export function Activity({ account, vault, showSwaps = true }: ActivityProps) {
   const paprController = useController();
-  const { data: swapsData, fetching: swapsFetching } = useUniswapSwapsByPool(
-    paprController.poolAddress,
-    account,
-  );
 
-  const { data: activityData, fetching: activityFetching } = useActivity(
-    paprController.id,
-    account,
-    vault,
-  );
+  const {
+    data: activityData,
+    fetching: activityFetching,
+    fetchMore,
+    remaining,
+  } = useActivity(paprController.id, account, vault, LIMIT);
 
-  const allEvents = useMemo(() => {
-    const unsortedEvents = [
-      ...(activityData?.addCollateralEvents || []),
-      ...(activityData?.removeCollateralEvents || []),
-      ...(swapsData?.swaps && showSwaps ? swapsData.swaps : []),
-      ...(activityData?.auctionStartEvents.filter(
-        (e) => e.auction.vault.controller.id === paprController.id,
-      ) || []),
-      ...(activityData?.auctionEndEvents.filter(
-        (e) => e.auction.vault.controller.id === paprController.id,
-      ) || []),
-    ];
-    return unsortedEvents.sort((a, b) => b.timestamp - a.timestamp);
-  }, [activityData, paprController.id, showSwaps, swapsData]);
-
-  const { feed, amountThatWillShowNext, remainingLength, showMore } =
-    useShowMore(allEvents, EVENT_INCREMENT);
-
-  if (swapsFetching || activityFetching) {
+  if (activityFetching && activityData.length === 0) {
     return <Fieldset legend="🐝 Activity">Loading...</Fieldset>;
   }
 
-  if (allEvents.length === 0) {
+  if (!activityFetching && activityData.length === 0) {
     return <Fieldset legend="🐝 Activity">No activity yet</Fieldset>;
   }
 
   return (
     <Fieldset legend="🐝 Activity">
       <Table>
         <tbody className={styles.table}>
-          {feed.map((event) => {
-            switch (event.__typename) {
-              case 'AddCollateralEvent':
+          {activityData.map((activity) => {
+            switch (true) {
```

Comment:
this is a bit of a peculiar way to do things -- I think this would actually be more readable with if statements